docs/adr/0001-update-logins-api.md

```diff
@@ -0,0 +1,96 @@
+    - [check_valid_with_no_dupes](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L663)
+    - [potential_dupes_ignoring_username](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L721)
+    - [import_multiple](https://github.com/mozilla/application-services/blob/1248a352cb2701b92395f2783bee8a88d18de455/components/logins/src/db.rs#L455)
+
+      **Note:** Functions related to sync have been omitted as it is assumed they will have access to decrypted data.
+
+  Propsed changes:
+    - Combine the `add` and `update` functions into a new `add_or_update` function
+      - This will allow the removal of consumer code that distinguishes when a login record should be created or updated
+      - **Note:** This function needs the encryption key for the fixup and deduping logic _and_ for continued support of the accurate population of the `time_password_changed` field
+    - Pass the per-field encryption key to the `import_multiple` function
+      - This function will be removed once the Fennec to Fenix migration period ends
+    - Remove both the `potential_dupes_ignoring_username` and `check_valid_with_no_dupes` from the API
+      - Neither function is called in Firefox iOS
+      - Android Components uses both to provide validation and de-duping before logins are added or updated so we can eliminate the need to externalize these functions by replicating this logic in the new `add_or_update` function
+    - Create a `decrypt_and_fixup_login` function that both decrypts a login _and_ performs the validate and fixup logic
+      - This will eliminate the need for the `get_all`, `get_by_base_domain`, and `get_by_id` API functions to perform the fixup logic
+
+    Making the above changes will reduce the API functions requiring the encryption key to the following:
+    - `add_or_update`
+    - `decrypt_and_fixup_login`
+    - `import_multiple`
+
+* Pros
+  * Improves the logins API for consumers by combining add/update functionality (see [#3899](https://github.com/mozilla/application-services/issues/3899) for details)
+  * Removes redundant validation and de-duping logic in consumer code
+  * Uses the same encryption model as autofill so there is consistency in our approaches
+* Cons
+  * Requires consumer code to both encrypt login fields _and_ pass the encryption key when calling either `add_or_update` and `import_multiple`
+
+
+### Option 2 - Implement a different key management approach
+
+* Description
+
+  With our current SQLCipher implementation, if the encryption key is not available when opening the database, the logins API is not accessible. We could replicate this logic with per-field encryption allowing us to store the encryption key for use with any calls that require it.
+
+  Proposed changes:
+  - Update the database layer logic to prevent accessing the database without an encryption key
```

Comment:
And here, we could tweak it to say "Any attempt to use the API will fail as the login records are encrypted or decrypted as the key isn't available" - I guess I'm kinda saying an explicit check probably isn't really necessary as the failures will happen implicitly.

Reply:
Since this isn't a feature that needs to be implemented, I'm going to remove the first bullet point and add, "Any attempt to use the API will fail as the login records aren't encrypted or decrypted if the key isn't available", to the description.